core/src/main/scala/org/apache/spark/ui/jobs/StagePage.scala

```diff
@@ -67,6 +68,8 @@ private[ui] class StagePage(parent: StagesTab) extends WebUIPage("stage") {
   // if we find that it's okay.
   private val MAX_TIMELINE_TASKS = parent.conf.getInt("spark.ui.timeline.tasks.maximum", 1000)
 
+  private val displayPeakExecutionMemory =
+    parent.conf.getOption("spark.sql.unsafe.enabled").exists(_.toBoolean)
```

Comment:
I haven't fully digested how this works yet, but won't you get this metric even w/out sql, and even w/out unsafe?  I thought that you are collecting the data on all uses of `Aggregator`?